### ISSUE TITLE
Does it make sense to move store project status in config instead of the db

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#3860]
D8 uses `core.extension.yml` to store info about which modules are enabled (or to be enabled, in the case you re moving config between environments). Would it make sense for us to move that out of the db and into config as well? Would there be performance benefit/hit (`module_exists()` etc.)? Any other things to consider?

**Comment by BWPanda** [comment]
Hmmm, good question!

**Comment by yorkshire-pudding** [comment]
I don't know about performance, but I think this would make it better when deploying changes.

**Comment by BWPanda** [comment]
@indigoxela Unless I'm misunderstanding, I think the only thing under discussion here is moving the list of enabled/disabled modules from the `system` table to a config file... I don't believe any of the install schemas, update hooks, etc. should/would change.

**Comment by indigoxela** [comment]
> I think the only thing under discussion here is moving the list of enabled/disabled modules from the system table to a config file

Exactly. And when it's in config, it gets moved along with other config exports/imports.

> I don't believe any of the install schemas, update hooks, etc. should/would change

Maybe you misunderstood my concern. It's not that the schema/update hooks change, but the imported config does not reflect the actual state of the current instance, but the state of another instance. Does that explain it better?

But maybe the other way round: what's the benefit of doing it "drupal8ish"? Is there anything cool in their approach, that Backdrop urgently needs?

**Comment by BWPanda** [comment]
> But maybe the other way round: what's the benefit of doing it "drupal8ish"?

Yes, I was wondering the same...

Also, what happens in a scenario where you have a dev site with, say, Antobot installed. Then you export config and push it up to production. Now Antibot is 'installed' in production, but the module doesn't actually exist in the codebase (i.e. it was never downloaded). Then what happens? Sure, we have a Project Installer module, so we could technically trigger that programmatically to go and download any missing 'installed' modules, but what if the Project Installer's not installed? How does Drupal handle that? I know they have Composer, but do Composer and `core.extension.yml` talk to each other?

**Comment by yorkshire-pudding** [comment]
I can imagine these applying whether the module enabled state is in the database or in config, but I imagine that code+config is more likely to be deployed between environments than database.

**Comment by keiserjb** [comment]
From the experience I've had Drupal sites break often when moving config from dev to live when modules are missing.  I then end up installing modules on live that I never intended to have there to get the site working.  I don't think I would vote for changing this in Backdrop.

**Comment by wilburec** [comment]
This trips me up when using backdrop almost every time I 'push' changes from local to prod.  You can capture everything else in code, but not the module list.  That has to be done manually.
